File: Data Structures & Algorithms/find-anagram-mappings/submission-0.py
class Solution:
    def anagramMappings(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res_dict = {}
        res = []

        for i in range(len(nums2)):
            res_dict[nums2[i]] = i
        
        for num in nums1: 
            res.append(res_dict[num])

        return res

File: Data Structures & Algorithms/find-the-difference-of-two-arrays/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums1
     * @param {number[]} nums2
     * @return {number[][]}
     */
    findDifference(nums1: number[], nums2: number[]): number[][] {
        const nums1Set = new Set(nums1);
        const nums2Set = new Set(nums2);

        const res1 = [];
        const res2 = [];

        for (const num of nums1Set) {
            if (!nums2Set.has(num)) {
                res1.push(num);
            }
        }

        for (const num of nums2Set) {
            if (!nums1Set.has(num)) {
                res2.push(num)
            }
        }

        return [res1, res2]
    }
}
